nixos/modules/services/monitoring/alertmanager-executor.nix

```diff
@@ -0,0 +1,70 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.alertmanager-executor;
+  settingsFormat = pkgs.formats.yaml {};
+
+in {
+  options = {
+    services.alertmanager-executor = {
+      enable = mkEnableOption "alertmanager-executor (aka prometheus-am-executor)";
+
+      user = mkOption {
+        type = types.str;
+        default = "alertmanager-executor";
+        description = lib.mdDoc "User to run the service and executed commands under.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "alertmanager-executor";
+        description = lib.mdDoc "Group to run the service and executed commands under.";
+      };
+
+      settings = mkOption {
+        type = settingsFormat.type;
+        default = {};
+        example = {
+          listen_address = "localhost:23222";
+          commands = [
+            {
+              cmd = "echo";
+              args = [ "alert" "firing" ];
+            }
+          ];
+        };
+        description = lib.mdDoc ''
+          Configuration for alertmanager-executor. Documentation:
+          <https://github.com/imgix/prometheus-am-executor#configuration-file-format>.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.users = optionalAttrs (cfg.user == "alertmanager-executor") {
+      alertmanager-executor = {
+        description = "Execute command based on Prometheus alerts";
+        isSystemUser = true;
+        group = cfg.group;
+      };
+    };
+
+    users.groups = optionalAttrs (cfg.group == "alertmanager-executor") {
+      alertmanager-executor = {};
+    };
+
+    systemd.services.alertmanager-executor = {
+      description = "Execute command based on Prometheus alerts";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "networking.target" ];
+      serviceConfig = {
+        ExecStart = "${pkgs.alertmanager-executor}/bin/prometheus-am-executor -v -f ${settingsFormat.generate "executor.yaml" cfg.settings}";
+        User = cfg.user;
+        Group = cfg.group;
+      };
```

Comment:
We need more hardening here: syscall filtering, namespaces, umask and etc.

Reply:
I've added some hardening options in [781541d](https://github.com/NixOS/nixpkgs/pull/189119/commits/781541d1c526b19c484f31b0020a9c75902f30b4). We can't know what NixOS users will configure as the commands to be executed so it's not possible to come up with the single set of options, though I suppose it's better to be overly restrictive and let the users override individual options in their config.

Reply:
It doesn't have to be cery comprehensive, just try poking around with `systemd-analyze security` and fix what sticks out the most. And yes, a user can always override the service definition.

Reply:
Added some more in [cb8b974](https://github.com/NixOS/nixpkgs/pull/189119/commits/cb8b9744d095bb3fc46178e529a98f0c6210e767)
```
Overall exposure level for alertmanager-executor.service: 1.5 OK
```